internal/command/deploy/machines_deploymachinesapp.go

```diff
@@ -311,8 +311,45 @@ type metadata struct {
 	value string
 }
 
+func (md *machineDeployment) guestForGroup(group string) *api.MachineGuest {
```

Comment:
hey, I still don't see the value of this function :( 

Starting with that it accepts a group name, but it is only called to spawn new machines (launchInputForLaunch) when there are no machines for that group. If there are no "siblings", which never are, it computes a median across all machines.

This is basically the same as computing such median in setMachineGuest(), and it is useful only if we agree the median is better than just defaulting to the minimum and deterministic shared-cpu-1x. 

Once the siblings and median idea is gone, the only left is basically what was before, start with `shared-cpu-1x` then apply vm-size, vm-cpus and vm-memory on top when set.  No need for `realizedGuest` either, just let `md.machineGuest` be a `*api.MachineGuest`

Reply:
> Starting with that it accepts a group name, but it is only called to spawn new machines (launchInputForLaunch) when there are no machines for that group. If there are no "siblings", which never are, it computes a median across all machines.

i don't think this is necessarily true! when service config changes, deploy might end up creating new machines in a group with existing machines, right?

Reply:
> i don't think this is necessarily true! when service config changes, deploy might end up creating new machines in a group with existing machines, right?

"deploy" only  creates machines when there are new groups, it doesn't if there are machines for exiting groups. 

The most common case is to update a machine, that won't change the machine guest.

And the last case is "replacing a machine" instead of updating it, this happens when volume config changes, but that reuses the original machine's guest.

hmmm.. it could be for canaries, need to check.